pkg/controller/statefulset/stateful_set_control.go

```diff
@@ -87,6 +87,14 @@ func (ssc *defaultStatefulSetControl) UpdateStatefulSet(set *apps.StatefulSet, p
 		return err
 	}
 
+	// maintain the set's revision history limit
+	defer func() {
```

Comment:
The issue with using a defer is that we cannot bubble up the error inside this block to the sync loop entry. 
It would introduce a subtle bug where in if truncateHistory fails, we would not requeue the sts for trying again. 

Would recommend not using a defer.

Reply:
Thanks for the comment. I didn't notice the requeue part and thanks for catching it!
I switched to another way instead of using `defer`. Alternatively, I invoke the `truncateHistory` function when any error occurs and return the combined errors.

FWIW, I used `go.uber.org/multierr` library. I checked `go.sum` and the library was there. I think we might be good to use it. But still want to double check with you @mortent @barney-s